src/libponyrt/sched/scheduler.c

```diff
@@ -577,6 +719,28 @@ void ponyint_sched_unnoisy_asio()
   send_msg_all(SCHED_UNNOISY_ASIO, 0);
 }
 
+// Maybe wake up a scheduler thread if possible
+void ponyint_sched_maybe_wakeup()
+{
+  uint32_t current_active_scheduler_count = get_active_scheduler_count();
+
+  bool cmp_val = false;
+
+  // if we have some schedulers that are sleeping, wake one up
+  if((current_active_scheduler_count < scheduler_count) &&
+    atomic_compare_exchange_strong_explicit(&scheduler_count_changing, &cmp_val,
```

Comment:
Same as in `steal`, exchange with acquire memory ordering.

Reply:
Thanks for the suggestion. I've changed the logic accordingly.